framework/include/materials/MaterialProperty.h

```diff
@@ -428,7 +428,13 @@ dataLoad(std::istream & stream, MaterialProperties & v, void * context)
   // Cast this to a vector so we can just piggy back on the vector store capability
   std::vector<PropertyValue *> & mat_props = static_cast<std::vector<PropertyValue *> &>(v);
 
+  // But the vector store capability should not be changing the vector
+  // size
+#ifndef NDEBUG
+  const std::size_t old_size = mat_props.size();
+#endif
   loadHelper(stream, mat_props, context);
+  libmesh_assert_equal_to(old_size, mat_props.size());
```

Comment:
Most typically you will see `mooseAssert` in MOOSE code but if you don't think you could hand-write an assertion error that adds more information than the stack trace, then I'm fine with this